actors/builtin/market/market_test.go

```diff
@@ -352,9 +354,289 @@ func TestPublishStorageDeals(t *testing.T) {
+		require.EqualValues(t, providerLocked, st.TotalProviderLockedCollateral)
+		totalStorageFee := big.Add(deal3.TotalStorageFee(), big.Add(deal1.TotalStorageFee(), deal2.TotalStorageFee()))
+		require.EqualValues(t, totalStorageFee, st.TotalClientStorageFee)
+
+		// publish two more deals for same clients with same provider
+		deal4 := generateDealProposal(client3, provider, abi.ChainEpoch(66), abi.ChainEpoch(70))
+		actor.addProviderFunds(rt, deal4.ProviderCollateral)
+		actor.addParticipantFunds(rt, client3, deal4.ClientBalanceRequirement())
+		deal5 := generateDealProposal(client3, provider, abi.ChainEpoch(101), abi.ChainEpoch(150))
+		actor.addProviderFunds(rt, deal5.ProviderCollateral)
+		actor.addParticipantFunds(rt, client3, deal5.ClientBalanceRequirement())
+		actor.publishDeals(rt, deal4, deal5)
+
+		// assert locked balances for clients and provider
+		rt.GetState(&st)
+		providerLocked = big.Add(providerLocked, big.Add(deal4.ProviderCollateral, deal5.ProviderCollateral))
+		require.EqualValues(t, providerLocked, actor.getLockedBalance(rt, provider))
+
+		client3LockedUpdated := actor.getLockedBalance(rt, client3)
+		require.EqualValues(t, big.Add(deal5.ClientBalanceRequirement(), big.Add(client3Locked, deal4.ClientBalanceRequirement())), client3LockedUpdated)
+
+		client1Locked = actor.getLockedBalance(rt, client1)
+		client2Locked = actor.getLockedBalance(rt, client2)
+		require.EqualValues(t, deal1.ClientBalanceRequirement(), client1Locked)
+		require.EqualValues(t, deal2.ClientBalanceRequirement(), client2Locked)
+
+		// assert locked funds states
+		totalClientCollateralLocked = big.Add(totalClientCollateralLocked, big.Add(deal4.ClientCollateral, deal5.ClientCollateral))
+		require.EqualValues(t, totalClientCollateralLocked, st.TotalClientLockedCollateral)
+		require.EqualValues(t, providerLocked, st.TotalProviderLockedCollateral)
+
+		totalStorageFee = big.Add(totalStorageFee, big.Add(deal4.TotalStorageFee(), deal5.TotalStorageFee()))
+		require.EqualValues(t, totalStorageFee, st.TotalClientStorageFee)
+	})
+}
+
+func TestPublishStorageDealsFailures(t *testing.T) {
+	marketActor := tutil.NewIDAddr(t, 100)
+	owner := tutil.NewIDAddr(t, 101)
+	provider := tutil.NewIDAddr(t, 102)
+	worker := tutil.NewIDAddr(t, 103)
+	client := tutil.NewIDAddr(t, 104)
+
+	// simple failures because of invalid deal params
+	{
+		tcs := map[string]struct {
+			setup                      func(*mock.Runtime, *marketActorTestHarness, *market.DealProposal)
+			exitCode                   exitcode.ExitCode
+			signatureVerificationError error
+		}{
+			"deal end after deal start": {
+				setup: func(_ *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					d.StartEpoch = 10
+					d.EndEpoch = 9
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"current epoch greater than start epoch": {
+				setup: func(rt *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					rt.SetEpoch(d.StartEpoch + 1)
```

Comment:
Rather than adjust the rt epoch somewhat haphazardly between tests, set the deal epoch relevant to the rt epoch, and leave the rt epoch fixed. That probably means you can use a constant for the rt epoch and remove `rt. from the signature here.

Reply:
This is done.